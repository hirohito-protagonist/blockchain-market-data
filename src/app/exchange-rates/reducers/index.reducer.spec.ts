import {
  getMarketPrices,
  getConvertBtcState,
  getCurrencies,
  viewExchangeRatesModel
} from './index.reducer';

describe('Exchange rates reducer', () => {

  describe('Selectors', () => {

    describe('getConvertBtcState', () => {

      it('should remap data service state to new model', () => {

        const selector = getConvertBtcState.projector({ isFetching: true, query: {}, lastUpdate: 123 });

        expect(selector).toMatchSnapshot();
      });

      it('should remap data service query state to new model', () => {

        const query = {
          value: '200',
          currency: 'USD',
        };
        const selector = getConvertBtcState.projector({ isFetching: true, query, response: 0.001, lastUpdate: 123 });

        expect(selector).toMatchSnapshot();
      });
    });

    describe('getCurrencies', () => {

      it('should return empty collection by default', () => {

        const selector = getCurrencies.projector({});

        expect(selector).toMatchSnapshot();
      });

      it('should remap market prices to new collection model', () => {

        const marketPrices = {
          'USD': { symbol: '$' },
          'PLN': { symbol: 'zł'}
        };
        const selector = getCurrencies.projector({ response: marketPrices });

        expect(selector).toMatchSnapshot();
      });
    });

    describe('viewExchangeRatesModel', () => {

      it('should compose selectors to create view model', () => {

        const marketPrices = {
          'USD': { symbol: '$' },
          'PLN': { symbol: 'zł'}
        };

        const prices = getMarketPrices.projector({ response: marketPrices });
        const currencies = getCurrencies.projector({ response: marketPrices });
        const convert = getConvertBtcState.projector({ isFetching: true, query: {}, response: 0.001, lastUpdate: 123 });
        const selector = viewExchangeRatesModel.projector(prices, currencies, convert);

        expect(selector).toMatchSnapshot();
      });
    });
  });
});
