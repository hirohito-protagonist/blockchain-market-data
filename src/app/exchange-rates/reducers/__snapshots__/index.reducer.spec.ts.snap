// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange rates reducer Selectors getConvertBtcState should remap data service query state to new model 1`] = `
Object {
  "convert": Object {
    "btc": 0.001,
    "currency": "USD",
    "value": "200",
  },
  "isFetching": true,
  "lastUpdate": 123,
}
`;

exports[`Exchange rates reducer Selectors getConvertBtcState should remap data service state to new model 1`] = `
Object {
  "convert": Object {
    "btc": 0,
    "currency": "",
    "value": "",
  },
  "isFetching": true,
  "lastUpdate": 123,
}
`;

exports[`Exchange rates reducer Selectors getCurrencies should remap market prices to new collection model 1`] = `
Array [
  Object {
    "currency": "USD",
    "symbol": "$",
  },
  Object {
    "currency": "PLN",
    "symbol": "zł",
  },
]
`;

exports[`Exchange rates reducer Selectors getCurrencies should return empty collection by default 1`] = `Array []`;

exports[`Exchange rates reducer Selectors viewExchangeRatesModel should compose selectors to create view model 1`] = `
Object {
  "data": Object {
    "convert": Object {
      "convert": Object {
        "btc": 0.001,
        "currency": "",
        "value": "",
      },
      "isFetching": true,
      "lastUpdate": 123,
    },
    "currencies": Array [
      Object {
        "currency": "USD",
        "symbol": "$",
      },
      Object {
        "currency": "PLN",
        "symbol": "zł",
      },
    ],
    "marketPrices": Object {
      "PLN": Object {
        "symbol": "zł",
      },
      "USD": Object {
        "symbol": "$",
      },
    },
  },
  "update": 123,
}
`;
